pandas/core/arrays/arrow/array.py

```diff
@@ -971,6 +974,27 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
         ------
         TypeError : subclass does not define reductions
         """
+        pa_type = self._data.type
+
+        if name in ["any", "all"] and (
+            pa.types.is_integer(pa_type)
+            or pa.types.is_floating(pa_type)
+            or pa.types.is_duration(pa_type)
+        ):
+            # pyarrow only supports any/all for boolean dtype, we allow
+            #  for other dtypes, matching our non-pyarrow behavior
+
+            zero: int | Timedelta
+            if pa.types.is_duration(pa_type):
+                zero = Timedelta(0)
+            else:
+                zero = 0
+            result = (self != zero)._reduce(name, skipna=skipna, **kwargs)
```

Comment:
Could we use `data = pc.not_equal(self._data, zero)` and continue with the logic below?